plugin/build.gradle

```diff
@@ -348,7 +352,9 @@ List<String> jacocoExclusions = [
         'org.opensearch.ml.helper.ModelAccessControlHelper',
         'org.opensearch.ml.action.models.DeleteModelTransportAction.2',
         'org.opensearch.ml.model.MLModelCacheHelper',
-        'org.opensearch.ml.model.MLModelCacheHelper.1'
+        'org.opensearch.ml.model.MLModelCacheHelper.1',
+        'org.opensearch.ml.sdkclient.SdkClientFactory',
```

Comment:
Is that a `TODO`?

Reply:
It's probably the local config causing coverage to drop, one more reason to remove it until it is properly implemented.

Reply:
It's also possible that it's not the local config, it's just the accumulation of lines of code on the Remote Client code branches, which aren't covered in unit tests.

I am strongly opposed to arbitrary minimum code coverage percentages and this is one reason why. The only alternative to maintainers is just to turn off the metric, rather than seek for maximum coverage. 90% isn't good enough if you've missed coverable lines but a 70% metric implies it is.  Sometimes 66% is all you can get without integ tests. Which I've written which cover all of these cases.

Reply:
Do you have any suggestion in mind?

Reply:
@dhrubo-os my general suggestion is to try to cover everything and evaluate missing coverage with an eye toward whether it is tested elsewhre (like integ tests).  I'm not a maintainer on this repo so it's not my place to overturn what you have in case; but the point here is that the more a class involves externalities (remote client etc.) the less you will be able to test it with unit tests, and the less meaning coverage will have.

Sure, you can write a bunch of mocks and "exercise" the method to artificially increase your coverage numbers, which doesn't mean the code actually works. :)

Ultimately my suggestion is to cover what you can and if it's not above an arbitrarily chosen metric, suppress it from failing like is done here.